### DISCUSSION TITLE
Organization of API-Keys

### DISCUSSION THREAD

**Discussion by iseneirik** [datalust/seq-tickets#2126] [Ideas]
Given the recommendation from the Docs on API-Keys:

>In general, every distinct application writing to Seq should have its own API key. If the application is deployed in multiple configurations, using a different key for each configuration is also a good idea.

It would be useful to have a way of organizing API-Keys.

A couple suggestions:
- Allow creation of folders
- Use the "Workspace"-feature to also separate API-Keys

**Comment by liammclennan** [comment]
Hi @iseneirik ,

Thanks for the suggestion. Can you please provide a little more detail about what you are trying to do? Why would it be useful to organize API-keys? Are you trying to find a particular API key? Are you thinking of the API KEYS page, the INGESTION page or both?

Thanks!

**Reply by iseneirik** [reply]
I was thinking mainly of the API KEYS page.

We have a number of applications, and quite a few environments where these applications run. This results in 3-6 API-keys per application. At the moment we use a naming convention to keep the API KEYS page clean, so we have `AppName - Environment`, which in turn allows us to use `AppName` in the search to quickly find keys.

I really like the idea of having many API-keys, as it allows a lot of configuration without having to redeploy or change config in the deployed applications, but some people find it messy.

So what I'm suggesting is adding collapsible folders to the API KEYS page, so instead of this:

```
ApplicationA - EnvironmentA
ApplicationA - EnvironmentB
ApplicationA - EnvironmentC
ApplicationB - EnvironmentA
ApplicationB - EnvironmentB
ApplicationB - EnvironmentC
ApplicationC - EnvironmentA
ApplicationC - EnvironmentB
```

It could be organized as:

```
ApplicationA/
├─ EnvironmentA
├─ EnvironmentB
├─ EnvironmentC
ApplicationB/
├─ EnvironmentA
├─ EnvironmentB
├─ EnvironmentC
ApplicationC/
├─ EnvironmentA
├─ EnvironmentB
```

For some cases it might even be usefull with deeper folders still (Application Group -> Application -> Environment/Config).

**Reply by liammclennan** [reply]
Understood. Thanks for the clarification.
